crates/bevy_dynamic_plugin/src/loader.rs

```diff
@@ -24,12 +24,16 @@ pub enum DynamicPluginLoadError {
 pub unsafe fn dynamically_load_plugin<P: AsRef<OsStr>>(
     path: P,
 ) -> Result<(Library, Box<dyn Plugin>), DynamicPluginLoadError> {
-    let lib = Library::new(path).map_err(DynamicPluginLoadError::Library)?;
-    let func: Symbol<CreatePlugin> = lib
-        .get(b"_bevy_create_plugin")
-        .map_err(DynamicPluginLoadError::Plugin)?;
-    let plugin = Box::from_raw(func());
-    Ok((lib, plugin))
+    // TODO: Safety comment for `Library::new`, `Library::get`, and `Box::from_raw`.
+    unsafe {
+        let lib = Library::new(path).map_err(DynamicPluginLoadError::Library)?;
+        let func: Symbol<CreatePlugin> = lib
+            .get(b"_bevy_create_plugin")
+            .map_err(DynamicPluginLoadError::Plugin)?;
+        let plugin = Box::from_raw(func());
+
+        Ok((lib, plugin))
+    }
```

Comment:
I think it's agains the idea of the lint to wrap the entire function body with the unsafe block.
IMHO, unsafe blocks should be as tiny as possible.

Reply:
I thought this too, but turns out there are three different unsafe functions being used here. The alternative would look like this:

```rust
let lib = unsafe { Library::new(path).map_err(DynamicPluginLoadError::Library)? };
let func: Symbol<CreatePlugin> = unsafe {
    lib.get(b"_bevy_create_plugin")
        .map_err(DynamicPluginLoadError::Plugin)?
};
let plugin = unsafe { Box::from_raw(func()) };

Ok((lib, plugin))
```

Which one do you prefer?

Reply:
The advantage of multiple unsafe blocks is that the safety comments are next to their call site, which I think is more clear than one big paragraph.

Reply:
That's a very good point. I split up the `unsafe` blocks and drafted safety comments for each. Thanks!